src/main/java/dev/j3fftw/litexpansion/machine/Generator.java

```diff
@@ -0,0 +1,143 @@
+package dev.j3fftw.litexpansion.machine;
+
+import dev.j3fftw.litexpansion.Items;
+import io.github.thebusybiscuit.slimefun4.implementation.items.electric.generators.CoalGenerator;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.MachineFuel;
+import org.bukkit.Material;
+import org.bukkit.Tag;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+
+public class Generator extends CoalGenerator {
+
+    public Generator() {
+        super(Items.LITEXPANSION, Items.GENERATOR, RecipeType.ENHANCED_CRAFTING_TABLE, new ItemStack[] {
+            null, Items.RE_BATTERY, null,
+            null, Items.MACHINE_BLOCK, null,
+            null, new ItemStack(Material.FURNACE), null
+        });
+    }
+
+    @Nonnull
+    @Override
+    public String getInventoryTitle() {
+        return "&7Generator";
+    }
+
+    @Nonnull
+    @Override
+    public ItemStack getProgressBar() {
+        return new ItemStack(Material.FIRE_CHARGE);
+    }
+
+    @Override
+    public int getEnergyProduction() {
+        return 10;
+    }
+
+    @Override
+    protected void registerDefaultFuelTypes() {
+
+        registerFuel(new MachineFuel(100, new ItemStack(Material.LAVA_BUCKET)));
```

Comment:
There is no output so people would use the bucket.

Reply:
im pretty sure AGenerator handles that

Reply:
Interesting, you are right, I thought you'd still have to specify the output manually.

Reply:
yea i just guessed XD